### ISSUE TITLE
Add ability to add a nested navigator push to the root navigator's backstack

### ISSUE THREAD

**Issue by carloshwa** [flutter/flutter#59193]
## Use case
If I push a route to a nested Navigator, that action does not get popped when I tap the device back button (Android) or the browser back button (Web). Would be nice to have the ability for those back buttons to pop whatever the last push was on any Navigator in the app.

## Proposal
Sample app:
https://github.com/carloshwa/flutter-example/tree/nested_navigator

This is based on the default sample app, but the FAB has been modified:
https://github.com/carloshwa/flutter-example/blob/nested_navigator/myapp/lib/main.dart#L58

to push a route to a nested Navigator to show the updated count:
https://github.com/carloshwa/flutter-example/blob/49cbe8e532233e9492574e4557be71097a707fad/myapp/lib/main.dart#L77-L86

After tapping the FAB, tapping the device back button or browser back button exits the app instead of navigating back to the previous count. Add the ability to have the back button pop the nested navigator's latest route (like Android's `FragmentTransaction.addToBackStack()` method).

**Comment by carloshwa** [comment]
After realizing I couldn't get this functionality, I tried to implement it using `addLocalHistoryEntry()` but then I ran into #59185. The only way I've been able to achieve this behavior is to push a new route to the root navigator that replaces the entire screen, which replaces shared widgets (like the drawer and other Scaffolds) that I would ideally like to leave untouched while replacing just a specific Scaffold.

**Comment by chunhtai** [comment]
The can be configured in the router widget in navigator 2.0.
For now, you can probably use [willPopScope](https://master-api.flutter.dev/flutter/widgets/WillPopScope-class.html)

**Comment by carloshwa** [comment]
Hi @chunhtai,
Thanks for the suggestion. I tried adding that to my sample app:
https://github.com/carloshwa/flutter-example/blob/nested_navigator/myapp/lib/main.dart#L71-L79

And it mostly achieves what I'm looking for, but it also has the same issue I reported in #59185 with using `addLocalHistoryEntry()` where in a web browser, the browser's back button does not get enabled when I push to one of the nested navigators, even though my `onWillPop` function would be able to perform a pop. So it seems I would also need a way to notify the root navigator that there is a stack available, so that it can enable the browser's back button.
